src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -185,26 +208,87 @@ public void start(final Map<String, String> props) {
     createPartitionMap(sourceConfig);
 
     MongoClientSettings.Builder builder =
-        MongoClientSettings.builder().applyConnectionString(sourceConfig.getConnectionString());
+        MongoClientSettings.builder()
+            .applyConnectionString(sourceConfig.getConnectionString())
+            .addCommandListener(
+                new CommandListener() {
+                  @Override
+                  public void commandSucceeded(final CommandSucceededEvent event) {
+                    String commandName = event.getCommandName();
+                    long elapsedTime = event.getElapsedTime(TimeUnit.NANOSECONDS);
+                    if ("getMore".equals(commandName)) {
+                      currentStatistics.successfulGetMoreCommand();
+                      currentStatistics.getMoreCommandElapsedTimeNanos(elapsedTime);
+                    } else {
```

Comment:
I don't expect any other commands (expect maybe killCursors?), but just to be a little defensive I would add a check here and below for "find" or "aggregate".